### ISSUE TITLE
Memory leak in IsSwapChainPanel

### ISSUE THREAD

**Issue by teshca** [microsoft/angle#102]
I noticed that once you call `CreateSurface(panel, ...)`
The ` panel ` will never be destroyed as well as all the listeners attached to it.
The leak takes place in a scenario where the page with angle surface located not on the MainPage of the application. So every time you go to PageWithAngle and back to MainPage you will leak instace of SwapChainPanel. I confirmed the leak with memory profiler tool in Visual Studio.

After an investiogation I found a leak in function `IsSwapChainPanel`
According to this article https://github.com/Microsoft/DirectXTK/wiki/ComPtr#constructing-assigning-and-copying-comptr

> If you want to give a pointer in one ComPtr to another ComPtr without changing the reference count (i.e. transferring ownership), you need to use Detach from the old ComPtr **followed by an Attach** to the new one.

```
bool IsSwapChainPanel(EGLNativeWindowType window, ComPtr<ABI::Windows::UI::Xaml::Controls::ISwapChainPanel> *swapChainPanel)
{
    if (!window)
    {
        return false;
    }

    ComPtr<IInspectable> win = window;
    ComPtr<ABI::Windows::UI::Xaml::Controls::ISwapChainPanel> panel;
    if (SUCCEEDED(win.As(&panel)))
    {
        if (swapChainPanel != nullptr)
        {
            *swapChainPanel = panel.Detach(); // <<-- LEAK IS HERE
        }
        return true;
    }

    return false;
}
```

So the Attach part is missing in the code above and this produces memory leak
The correct way to achive the result would be
`swapChainPanel->Attach(panel.Detach());`

Same thing about `IsCoreWindow`

**Comment by austinkinross** [comment]
Thanks for the report! We'll take a look at this soon.

**Comment by abucur** [comment]
We've also noticed memory leaks from ANGLE in our multi-window application. They hit us pretty bad because the SwapChainPanel is leaking the D3D device as well. Fortunately, calling Window::Close() on the secondary windows seems to somehow cleanup the leakage due to how CDependencyObject classes are implemented. Unfortunately, the main window will always leak. Kudos @teshca for finding it!

**Comment by austinkinross** [comment]
Thanks again for reporting this! The issue has been fixed in both master and ms-master. The fix is also available in the latest NuGet package (2.1.12).
